zerver/lib/email_notifications.py

```diff
@@ -163,34 +163,24 @@ def build_sender_payload(message: Message) -> Dict[str, Any]:
                 'content': [build_message_payload(message)]}
 
     def message_header(user_profile: UserProfile, message: Message) -> Dict[str, Any]:
+        narrow_link = get_narrow_url(user_profile, message)
         if message.recipient.type == Recipient.PERSONAL:
             header = "You and %s" % (message.sender.full_name,)
-            html_link = personal_narrow_url(
-                realm=user_profile.realm,
-                sender=message.sender,
-            )
-            header_html = "<a style='color: #ffffff;' href='%s'>%s</a>" % (html_link, header)
+            header_html = "<a style='color: #ffffff;' href='%s'>%s</a>" % (narrow_link, header)
         elif message.recipient.type == Recipient.HUDDLE:
             disp_recipient = get_display_recipient(message.recipient)
             assert not isinstance(disp_recipient, str)
             other_recipients = [r['full_name'] for r in disp_recipient
                                 if r['id'] != user_profile.id]
             header = "You and %s" % (", ".join(other_recipients),)
-            other_user_ids = [r['id'] for r in disp_recipient
-                              if r['id'] != user_profile.id]
-            html_link = huddle_narrow_url(
-                realm=user_profile.realm,
-                other_user_ids=other_user_ids,
-            )
-
-            header_html = "<a style='color: #ffffff;' href='%s'>%s</a>" % (html_link, header)
+            header_html = "<a style='color: #ffffff;' href='%s'>%s</a>" % (narrow_link, header)
         else:
             stream = Stream.objects.only('id', 'name').get(id=message.recipient.type_id)
             header = "%s > %s" % (stream.name, message.topic_name())
             stream_link = stream_narrow_url(user_profile.realm, stream)
-            topic_link = topic_narrow_url(user_profile.realm, stream, message.topic_name())
+            narrow_link = topic_narrow_url(user_profile.realm, stream, message.topic_name())
```

Comment:
This seems like a redundant call with what's above and should be deleted?  Might explain the surprising increase in number of database queries in this commit.